hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/streaming/HoodieSourceOffset.scala

```diff
@@ -45,17 +45,21 @@ case class HoodieSourceOffset(commitTime: String) extends Offset {
 
 
 object HoodieSourceOffset {
-  val mapper = new ObjectMapper with ScalaObjectMapper
-  mapper.setSerializationInclusion(Include.NON_ABSENT)
-  mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-  mapper.registerModule(DefaultScalaModule)
+
+  lazy val mapper: ObjectMapper = {
+    val _mapper = new ObjectMapper
+    _mapper.setSerializationInclusion(Include.NON_ABSENT)
+    _mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    _mapper.registerModule(DefaultScalaModule)
+    _mapper
+  }
 
   def toJson(offset: HoodieSourceOffset): String = {
     mapper.writeValueAsString(offset)
   }
 
   def fromJson(json: String): HoodieSourceOffset = {
-    mapper.readValue[HoodieSourceOffset](json)
+    mapper.readValue(json, classOf[HoodieSourceOffset])
```

Comment:
is the change necessary?

Reply:
After remove `ScalaObjectMapper`, we need `ClassTag` to readValue.